File: lms-BE/app/features/activity_logs/schemas.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class UserLogInfo(BaseModel):
    id: int
    name: str
    email: str
    role: str

    class Config:
        from_attributes = True

class ActivityLogCreate(BaseModel):
    user_id: Optional[int] = None
    action: str
    entity_type: Optional[str] = None
    entity_id: Optional[int] = None
    details: Optional[str] = None

class ActivityLogRead(BaseModel):
    id: int
    user_id: Optional[int] = None
    action: str
    entity_type: Optional[str] = None
    entity_id: Optional[int] = None
    details: Optional[str] = None
    created_at: datetime
    
    user: Optional[UserLogInfo] = None

    class Config:
        from_attributes = True

class PaginatedActivityLogs(BaseModel):
    items: list[ActivityLogRead]
    total: int
    page: int
    size: int
    pages: int

File: lms-FE/src/features/activityLogs/schemas.ts
import { z } from 'zod';

const userLogInfoSchema = z.object({
    id: z.number(),
    name: z.string(),
    email: z.string(),
    role: z.string(),
});

export const activityLogSchema = z.object({
    id: z.number(),
    user_id: z.number().nullable(),
    action: z.string(),
    entity_type: z.string().nullable(),
    entity_id: z.number().nullable(),
    details: z.string().nullable(),
    created_at: z.string(),
    user: userLogInfoSchema.nullable(),
});

export const paginatedActivityLogsSchema = z.object({
    items: z.array(activityLogSchema),
    total: z.number(),
    page: z.number(),
    size: z.number(),
    pages: z.number(),
});

export type ActivityLog = z.infer<typeof activityLogSchema>;
export type PaginatedActivityLogs = z.infer<typeof paginatedActivityLogsSchema>;
